meeshkan/serve/commands.py

```diff
@@ -66,35 +67,45 @@ def _add_options(func):
         type=click.Path(exists=True, file_okay=False, resolve_path=True),
         help="Directory with callback scripts to modify behavior.",
     ),
+    click.option("-k", "--kill", is_flag=True, help="Stop the mock daemon."),
     click.option(
         "-l",
         "--log-dir",
         default=None,
         type=click.Path(exists=False, file_okay=False, resolve_path=True),
         help="Directory where server logs are written.",
     ),
+    click.option(
+        "-s", "--status", is_flag=True, help="Show the status of the mock daemon."
+    ),
 ]
 
 
-@click.group(invoke_without_command=True)
+@click.command()
 @add_options(_mock_options)
-@click.pass_context
+@click.argument("specifications", nargs=-1)
 def mock(
-    ctx, callback_dir, log_dir, admin_port, port, specs_dir, header_routing, daemon
+    callback_dir,
+    log_dir,
+    admin_port,
+    port,
+    header_routing,
+    daemon,
+    kill,
+    status,
+    specifications,
 ):
     """
     Run a mock server using OpenAPI specifications.
     """
-    if ctx.invoked_subcommand is None:
-        ctx.forward(start_mock)
-
+    if kill:
+        stop_mocking()
+        sys.exit(0)
+    elif status:
+        status_mocking()
+        sys.exit(0)
 
-@mock.command(name="start")  # type: ignore
-@add_options(_mock_options)
-def start_mock(
-    callback_dir, log_dir, admin_port, port, specs_dir, header_routing, daemon
-):
-    specs = load_specs(specs_dir)
+    specs = load_specs(specifications)
```

Comment:
I think a positional argument requires quite good validation for the input, is that done my `load_specs` method? I think it might be clearer if the validation was done here explicitly?

Reply:
The validation is done in `load_specs`. It now throws an `Exception`, which we in `serve/commands.py` wrap into a `ClickException`. I think that works fine, but I'm open to suggestions for how to split up the handling!